### ISSUE TITLE
When using TypeORM with Postgres, failed connection errors cannot be caught.

### ISSUE THREAD

**Issue by rajshrimohanks** [typeorm/typeorm#658]
I made a sample app here that demonstrates the issue: [TypeORM `createConnection()` catch fail](https://github.com/rajshrimohanks/typeorm-connection-catch-fail)

When using `createConnection()` method to create a connection, if the credentials and connection is good, the method succeeds and we are able to get the `connection` object in the promise resolution. However, if the credentials are wrong or if the connection times out, the error thrown does not get caught in the `catch` method of the Promise. Instead, an `UnhandledPromiseRejectionWarning` is thrown in the console:

**This happens when using `pg`. With `mysql` everything works fine.**

### Sample Code
```typescript
// Import stuff required by TypeORM to initialize DB connection.
import 'reflect-metadata';
import { createConnection } from 'typeorm';

createConnection({
    host: 'localhost',
    port: 5432,
    username: 'postgres',
    password: 'samplepass',
    database: 'sampledb',
    type: 'postgres',
    entities: [
        __dirname + '/models/{*.ts,*.js}'
    ],
    autoSchemaSync: true
}).then(connection => {
    console.log('Connected!');
}).catch(error => {
    console.error('Error!');
});
```
### Expected Output
```
Error!
```
### Actual Output
```
(node:714) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: connect ECONNREFUSED 127.0.0.1:5432
(node:714) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
### Package Versions Used
- `typeorm`: 0.1.0-alpha.31
- `pg`: 6.4.0
- `typescript`: 2.4.1
- `reflect-metadata`: 0.1.10

**Comment by pleerock** [comment]
Thank you for your report. Fix is released in `0.1.0-alpha.32`

**Comment by jayarjo** [comment]
@pleerock  I'm encountering the same error in `0.2.20`: https://github.com/typeorm/typeorm/issues/5019
